### ISSUE TITLE
[java] CloseResource false positive with Objects.nonNull

### ISSUE THREAD

**Issue by dpalic** [pmd/pmd#3148]
**Affects PMD Version:** 
6.30.0

**Rule:**

https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource

**Description:**

if using 
`if (Objects.nonNull(preparedStmt))`
instead of
`if (preparedStmt != null)`
rule still fires a violation

**Code Sample demonstrating the issue:**
```
      PreparedStatement lPreparedStmt = null;
      try {
         lPreparedStmt = conn.prepareStatement(sqL);
         lPreparedStmt.execute();
      } catch (SQLException ex) {
         LOG.debug("lPreparedStmt.execute();loooooooooooooose" + ex);
      } finally {
         if (Objects.nonNull(lPreparedStmt)) {
            try {
               lPreparedStmt.close();
            } catch (SQLException pEx) {
               LOG.error("unrecoverable", pEx);
            }
         }
      }

```

**Expected outcome:**

false-positive

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
eclipse acanda plugin

**Comment by adangel** [comment]
Thanks for reporting. I can confirm, that's a false positive. We seem not to detect the close call, if it is inside a if-block.
We might have some logic there, because if the condition is not as simple as "!= null", we can't be sure, that it is actually executed.

Btw. - if you can use try-with-resources, use that, instead of a finally block.

**Comment by adangel** [comment]
I can only reproduce the problem when using `Objects.nonNull` - which we definitively do not cover and which could be added. But when I use `lPreparedStmt != null`, the rule works and doesn't report a violation.

Here's my complete example which doesn't report a violation:

```java
import java.sql.*;

public class FalsePositive {

    public void bar(Connection conn, String sqL) {
        PreparedStatement lPreparedStmt = null;
        try {
            lPreparedStmt = conn.prepareStatement(sqL);
            lPreparedStmt.execute();
        } catch (SQLException ex) {
            System.out.println("lPreparedStmt.execute();loooooooooooooose" + ex);
        } finally {
            if (lPreparedStmt != null) {
                try {
                    lPreparedStmt.close();
                } catch (SQLException pEx) {
                    System.err.println("unrecoverable:" + pEx);
                }
            }
        }
    }
}
```

@dpalic Could you please double check?
